klippy/extras/gcode_move.py

```diff
@@ -256,8 +256,11 @@ def cmd_GET_POSITION(self, gcmd):
         kin_pos = " ".join(["%s:%.6f" % (a, v) for a, v in kinfo])
         toolhead_pos = " ".join(["%s:%.6f" % (a, v) for a, v in zip(
             "XYZE", toolhead.get_position())])
+        last_pos = self.last_position
+        for pos in range(len(last_pos)):
+            last_pos[pos] = last_pos[pos] - self.base_position[pos]
         gcode_pos = " ".join(["%s:%.6f"  % (a, v)
-                              for a, v in zip("XYZE", self.last_position)])
```

Comment:
I think you just need to change this line to 
```
                              for a, v in zip("XYZE", self._get_gcode_position())])
```

Reply:
This is a good idea! Not only it is simpler, it will also report correct E position (accounting for self.extrude_factor)

Reply:
This is much better than the original fix!
Thank you!